### PR TITLE
feat: Enable ACH via Stripe Vaulting flows

### PR DESCRIPTION
**Pull request by NQuinn27**
# Description

[ACC-4384](https://primerapi.atlassian.net/browse/ACC-4384)
[ACC-4461](https://primerapi.atlassian.net/browse/ACC-4461)
[ACC-4515](https://primerapi.atlassian.net/browse/ACC-4515)

- Adds vaultOnAgreement to the Debug App UI
- Adds necessary deviceInfo to metadata for ACH flows
- Show vaulted ACH payment methods in Drop-in UI
- Enable paying with a vaulted ACH payment method on both Drop-in and Headless

# Other Notes

# Manual Testing

_Add manual testing notes here if applicable, otherwise remove this section_

# Screenshots

_If applicable, otherwise remove this section_

# Contributor Checklist

- [ ]  All status checks have passed prior to code review
- [ ]  I have added unit tests to a reasonable level of coverage where suitable
- [ ]  I have added UI tests to new user flows, if applicable
- [ ]  I have manually tested newly added UX
- [ ]  I have open a documentation PR, if applicable

# Reviewer Checklist

- [ ]  I have verified that a suitable set of automated tests has been added
- [ ]  I have verified that the title prefix aligns to the code changes + whether a release is expected after merging the PR
- [ ]  I have verified the documentation PR aligns with this PR, if applicable

# Before Merging

- [ ]  If introducing a breaking change, I have communicated it internally
- [ ]  Any related documentation PRs are ready to merge

# Other Stuff

- You can find out more about our automation checks [here](https://primerio.notion.site/iOS-Automation-Checks-198a1eb0e8994d999fb696d5902d97bb)
- Find out more about conventional commits [here](https://primerio.notion.site/Conventional-Commits-6ecfd6a0269a4db2af76d9f0537936b3)

[ACC-4384]: https://primerapi.atlassian.net/browse/ACC-4384?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ACC-4461]: https://primerapi.atlassian.net/browse/ACC-4461?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ACC-4515]: https://primerapi.atlassian.net/browse/ACC-4515?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by borisprimer — APPROVED** [review]
LGTM.

**Review by borisprimer — DISMISSED** [review]

File: Sources/PrimerSDK/Classes/Core/Payment Services/CreateResumePaymentService.swift [hunk 1]
```diff
@@ -97,4 +98,34 @@ internal class CreateResumePaymentService: CreateResumePaymentServiceProtocol {
...
+     * and a URL indicating where the completion request should be sent.
+     *
+     * - Parameters:
+     *   - clientToken: A `DecodedJWTToken` representing the client's authentication token.
+     *   - completeUrl: An `URL` indicating the endpoint for completing the ACH payment.
+     *
+     * - Returns: A `Promise<Void>` that resolves if the payment is completed successfully, or rejects if there is
+     *            an error during the API call.
+     */
+    func completePayment(clientToken: DecodedJWTToken, completeUrl: URL) -> Promise<Void> {
```
Comment (semirp) [inline Sources/PrimerSDK/Classes/Core/Payment Services/CreateResumePaymentService.swift:115]: [q]: wouldn't it make more sense to send some params to be mapped to request body or maybe request body itself? This seems to be a generic function that creates body for only one use case.

**Review by semirp — APPROVED** [review]
PCI OK.

**Review by semirp — APPROVED** [review]
PCI OK

**Review by semirp — APPROVED** [review]
PCI OK

**Review by borisprimer — APPROVED** [review]

**Pull request merged**